programs/mmm/src/instructions/mip1/sol_mip1_fulfill_sell.rs

```diff
@@ -76,25 +76,27 @@ pub struct SolMip1FulfillSell<'info> {
     pub asset_metadata: UncheckedAccount<'info>,
     #[account(
         constraint = asset_mint.supply == 1 && asset_mint.decimals == 0 @ MMMErrorCode::InvalidMip1AssetParams,
+
     )]
-    pub asset_mint: Account<'info, Mint>,
+    pub asset_mint: InterfaceAccount<'info, Mint>,
     /// CHECK: will be checked in cpi
     pub asset_master_edition: UncheckedAccount<'info>,
     #[account(
         mut,
         associated_token::mint = asset_mint,
         associated_token::authority = pool,
+        associated_token::token_program = token_program,
         constraint = sellside_escrow_token_account.amount == 1 @ MMMErrorCode::InvalidMip1AssetParams,
         constraint = args.asset_amount == 1 @ MMMErrorCode::InvalidMip1AssetParams,
     )]
-    pub sellside_escrow_token_account: Box<Account<'info, TokenAccount>>,
+    pub sellside_escrow_token_account: Box<InterfaceAccount<'info, TokenAccount>>,
     #[account(
         init_if_needed,
         associated_token::mint = asset_mint,
         associated_token::authority = payer,
```

Comment:
no need for associated_token::token_program here?

Reply:
I'm not actually sure why the tests are passing here without this but I added it.